website/project/model.py

```diff
@@ -3691,3 +3770,140 @@ def _on_reject(self, user, token):
+
+    _id = fields.StringField(primary=True, default=lambda: str(ObjectId()))
+
+    datetime_initiated = fields.DateTimeField(auto_now_add=True)
+    datetime_updated = fields.DateTimeField(auto_now=True)
+    # Original Node a draft registration is associated with
+    branched_from = fields.ForeignField('node')
+
+    initiator = fields.ForeignField('user')
+
+    # Dictionary field mapping question id to a question's comments and answer
+    # {
+    #   <qid>: {
+    #     'comments': [{
+    #       'user': {
+    #         'id': <uid>,
+    #         'name': <name>
+    #       },
+    #       value: <value>,
+    #       lastModified: <datetime>
+    #     }],
+    #     'value': <value>
+    #   }
+    # }
+    registration_metadata = fields.DictionaryField(default=dict)
+    registration_schema = fields.ForeignField('metaschema')
+    registered_node = fields.ForeignField('node')
+
+    # TODO (samchrisinger): It would be better to run Archiver tasks when a new
+    # DraftRegistration gets created. Files could be copied to a DraftRegistration
+    # rather than a Node, which would be much much cleaner in the event of a
+    # failure during archival.
+    # Additionally, future registration schemas will require users to select files
+    # that fulfill a certain requirement. Right now we will have to restrict those
+    # choices to OsfStorage files, but this is a non-issue if the third-party files
+    # have already been archived.
+    # storage = fields.ForeignField('osfstoragenodesettings')
+
+    approval = fields.ForeignField('draftregistrationapproval', default=None)
+
+    # Dictionary field mapping extra fields defined in the MetaSchema.schema to their
+    # values. Defaults should be provided in the schema (e.g. 'paymentSent': false),
+    # and these values are added to the DraftRegistration
+    flags = fields.DictionaryField()
+
+    notes = fields.StringField()
+
+    def __init__(self, *args, **kwargs):
+        super(DraftRegistration, self).__init__(*args, **kwargs)
+        meta_schema = self.registration_schema or kwargs.get('registration_schema')
+        if meta_schema:
+            schema = meta_schema.schema
+            if not self.registration_schema:
+                flags = schema.get('flags', {})
+                for flag, value in flags.iteritems():
+                    self.flags[flag] = value
+
+    @property
+    def requires_approval(self):
+        return self.registration_schema.requires_approval
+
+    @property
+    def is_pending_review(self):
+        return self.approval.is_pending_approval if (self.requires_approval and self.approval) else False
+
+    @property
+    def is_approved(self):
+        if self.requires_approval and not self.approval:
+            return False
+        elif self.requires_approval and self.approval:
+            return self.approval.is_approved
+        else:
+            return True
+
+    def update_metadata(self, metadata, save=True):
```

Comment:
The existing convention (eg the majority of Node model methods) is to avoid implicit saves, so that changing the database is something that has to be consciously declared: `save=False` as default value for this kwarg and all usages updated.